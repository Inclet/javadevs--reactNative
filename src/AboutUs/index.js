import React from "react";
import { View, StyleSheet, Text, Image, TouchableOpacity } from "react-native";
import theme from "../Styles/theme.style";
import common from "../Styles/common.style";
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp
} from "react-native-responsive-screen";
import AboutImage from "../Assets/Images/webdesign.png";
import Layout from "../Common/Layout/index";
import CommonButton from "../Common/Button/index";
const AboutUs = ({ navigation }) => {
  return (
    <Layout target={navigation} currentPage={2}>
      <View style={common.innerPageLayout}>
        <Image source={AboutImage} style={styles.aboutImageStyle} />
        <Text style={common.brandStyle}>JavaDevs</Text>
        <Text style={styles.sloganStyle}>We are in this together</Text>
        <Text style={styles.descriptionStyle}>
          Choose a java developer from a large pool of world-class developers.
        </Text>
        <CommonButton
          target={navigation}
          label="Let's Go"
          newMarginTop={80}
          page="AuthScreen"
        />
      </View>
    </Layout>
  );
};

export default AboutUs;

const styles = StyleSheet.create({
  aboutImageStyle: {
    width: wp("100%"),
    height: hp("30%"),
    marginTop: 50
  },
  sloganStyle: {
    fontSize: theme.FONT_SIZE_MEDIUM,
    color: theme.SECONDARY_COLOR,
    marginTop: 15
  },
  descriptionStyle: {
    fontSize: theme.FONT_SIZE_SMALL,
    color: theme.MEDIUM_COLOR,
    width: wp("80%"),
    textAlign: "center",
    marginTop: 30
  }
});
