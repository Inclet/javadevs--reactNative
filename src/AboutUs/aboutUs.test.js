import React from "react";
import { shallow } from "enzyme";
import AboutUs from "./index";

describe("About Us Screen", () => {
  let wrapper, props;
  beforeEach(() => {
    props = {
      navigation: {
        navigate: jest.fn()
      }
    };
    wrapper = shallow(<AboutUs {...props} />);
  });
  it("Should match Snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("Should Navigate to Auth screen when clicking on let's go button", () => {
    wrapper.find('[data-testID="authScreenNavigator"]').simulate("press");
    expect(props.navigation.navigate).toHaveBeenCalledWith("AuthScreen");
  });
});
