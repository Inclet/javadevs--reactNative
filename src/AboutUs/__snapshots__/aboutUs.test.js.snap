// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Us Screen RENDERING Should match Snapshot 1`] = `ShallowWrapper {}`;
